src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+    ptr->flags = 0;
+
+    *iconf = ptr;
+}
+
+ConfNode *ConfSetRootIfaceNode(const char *dpdk_node_name, const char *iface)
+{
+    ConfNode *if_root;
+    ConfNode *dpdk_node;
+    /* Find initial node */
+    dpdk_node = ConfGetNode(dpdk_node_name);
+    if (dpdk_node == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find %s config", dpdk_node_name);
+    }
+
+    if_root = ConfFindDeviceConfig(dpdk_node, iface);
+    if (if_root == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find interface %s in DPDK config", iface);
+    }
+
+    return if_root;
+}
+
+ConfNode *ConfSetDefaultNode(const char *dpdk_node_name)
+{
+    ConfNode *if_default;
+    ConfNode *dpdk_node;
+    /* Find initial node */
+    dpdk_node = ConfGetNode(dpdk_node_name);
+    if (dpdk_node == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find %s config", dpdk_node_name);
+    }
+
+    if_default = ConfFindDeviceConfig(dpdk_node, "default");
+    if (if_default == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find default interface in DPDK config");
+    }
+
+    return if_default;
+}
+
+int ConfSetRootAndDefaultNodes(
+        const char *dpdk_node_name, const char *iface, ConfNode **if_root, ConfNode **if_default)
+{
+    *if_root = ConfSetRootIfaceNode(dpdk_node_name, iface);
+    *if_default = ConfSetDefaultNode(dpdk_node_name);
+
+    if (*if_root == NULL && *if_default == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF,
+                "unable to find DPDK config for "
+                "interface \"%s\" or \"default\", using default values",
+                iface);
+        SCReturnInt(-ENODEV);
+    }
+
+    /* If there is no setting for current interface use default one as main iface */
+    if (*if_root == NULL) {
+        *if_root = *if_default;
+        *if_default = NULL;
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetIface(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    int retval;
+
+    if (entry_str == NULL || entry_str[0] == '\0') {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    retval = rte_eth_dev_get_port_by_name(entry_str, &iconf->port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Name of the interface (%s) not valid", entry_str);
+        errno = -retval;
+        SCReturnInt(retval);
+    }
+
+    strlcpy(iconf->iface, entry_str, sizeof(iconf->iface));
+    SCReturnInt(0);
+}
+
+int ConfigSetThreads(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    const char *active_runmode = RunmodeGetActive();
+
+    if (active_runmode && !strcmp("single", active_runmode)) {
+        iconf->threads = 1;
+        SCReturnInt(0);
+    }
+
+    if (entry_str == NULL) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (strcmp(entry_str, "auto") == 0) {
+        iconf->threads = (int)rte_lcore_count() - 1;
```

Comment:
since you removing 1 lcore (master core) from the total count, are you running statistics on master core?